programs/gso/src/instructions/unstake.rs

```diff
@@ -7,9 +7,17 @@ pub use crate::*;
 pub fn unstake(ctx: Context<GSOUnstake>, amount: u64) -> Result<()> {
     msg!("GSO Unstake");
     let now_ts: u64 = Clock::get().unwrap().unix_timestamp as u64;
-    let expiration: u64 = ctx.accounts.gso_state.subscription_period_end;
-    msg!("Now {} Expiration {}", now_ts, expiration);
-    invariant!(expiration < now_ts, NotYetExpired);
+    let expiration: u64 = ctx.accounts.gso_state.lockup_period_end;
+    if expiration > 0 {
```

Comment:
Old GSO's have an undefined or null lockup period?

Reply:
They dont have anything. It is undefined behavior what will be at those bytes. It is possible that this change will make it impossible to unstake from any current GSO.

Reply:
I have a bunch of GSO ready to be unstaked on mainnet I can test once deployed